pex/cli/commands/lock.py

```diff
@@ -1414,8 +1428,28 @@ def _sync(self):
                     pip_configuration=pip_configuration,
                 )
             )
-            with safe_open(lock_file_path, "w") as fp:
-                self._dump_lockfile(lockfile, output=fp)
+            if self.options.dry_run:
```

Comment:
This seems like an appropriate way to handle creating with `--dry-run`, thanks for getting it in.